Design system

chrisdel101 opened the pull request:
Reallizes #1628.
- moving colors to variables. Attempting to make colors/names constant across themes.
- move language css to folder
- move themes to folder
- remove unused code them applicable
- add light mode theme

Left some comments in the codes to note why changes were made. These need deletion before merge. 
More commentary on reasons for changes given below.

preview: https://deploy-preview-1742--expressjscom-preview.netlify.app/

Commit:
Adding css to folders
- theme files and import

Commit:
Add main vars file + add light mode
- add dark mode announcment icon
- toggle light mode class on/off
- work on vars name and vals

Commit:
Merge branch 'gh-pages' of https://github.com/expressjs/expressjs.com into design-system

Commit:
Add links for lightm and darkm
- add lightm grey side menu color
- add global gray text color

Commit:
Move all lightm vars out of styles
- mirror moved styles in darkm
- note num vals to new vars

Commit:
Doctable table head and card
- lightm lightens the grey
- darkm ligher black
- move global greys to each scope
 - add darkm scroll styles - undo at end

Commit:
Fix pre plain-text color
- move menu hover boxshadows to themes
- api #menu colors

Commit:
Move doc boxes to themes

Commit:
blog, eol and support classes

Commit:
Move styles back into styles using same var for both
- realign into original position
- work on and up to pre tag section

Commit:
Contiue moving styles back into style.css

Commit:
Finish moving styles back
- redo nav menus

Commit:
Move search bar theme to styles

Commit:
Remove all items possible from themes to style.css

Commit:
Add few more inline comments

chrisdel101 commented on _includes/head.html:
```diff
@@ -5,12 +5,13 @@
     <link rel="stylesheet" href="/css/font-awesome.min.css">
     <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Open+Sans:300,400,600,700&amp;amp;subset=latin,latin-ext">
-    <link rel="stylesheet" href="/css/{{ page.lang }}.css">
+    <link rel="stylesheet" href="/css/langs/{{ page.lang }}.css">
```
Move these files into folders.

bjohansebas replied:
If we end up not migrating, i would like to delete all those files

chrisdel101 commented on _layouts/home.html:
```diff
@@ -3,7 +3,7 @@
  License: MIT
 -->
-<html lang="{{ page.lang }}">
+<html class="light-mode" lang="{{ page.lang }}">
```
Add light-mode by default. Some JS was added to add remove the class, along with dark-mode.

bjohansebas replied:
Why not do it globally? I don't see the need to add it to just one class.

chrisdel101 replied:
The css is triggered by light-mode or dark-mode now, so it must have a theme class on the html. This just the default - open to change if it's possible. What did you have in mind?

bjohansebas replied:
```suggestion
<html lang="{{ page.lang }}">
```

Not adding default classes, but well, I see it's used elsewhere, so we would need to debug further. I'll be back soon, I'll keep removing prims from the page.

chrisdel101 replied:
We need someway to trigger the themes. It's pretty well based around this setup now, and is what allows the vars to work mostly in one file - style.css. Don't know it can be unwound - unless you see a way. 
Anyway, it's just a single class so the overhead is very low. And the JS used is very simple, add remove class stuff

bjohansebas replied:
Yep, I can live with that.

chrisdel101 replied:
I'm testing it now that you say this to make sure. It might be added by the JS already so this class is _not_ needed. If this is the case then yes I'll remove it. Just need to test to make sure.
I removed the class and it still works, so...

chrisdel101 replied:
Removed

chrisdel101 commented on css/themes/dark-theme.css:
```diff
@@ -0,0 +1,80 @@
+  }
+
+  /* overlay on dropdown menu  */
+  div#overlay {
```
Changed the overlay to be more pronounced
![1_dark_mode_overlay_color](https://github.com/user-attachments/assets/6ab1dca6-b280-4d91-a96c-c2df60b2d51a)

bjohansebas replied:
great

chrisdel101 commented on css/themes/light-theme.css:
```diff
@@ -0,0 +1,15 @@
+        background: var(--box-fg);
+    }
+
+    #overlay {
```
Add more pronounced overlay.
![1_light_mode_overlay_color](https://github.com/user-attachments/assets/71d20619-ff39-4c72-a4cf-7cd0f4becb93)

bjohansebas replied:
great

chrisdel101 commented on css/style.css:
```diff
@@ -320,72 +302,102 @@ header {
+/* all pre tags instances have code children - get rid of importants - override prism white bg */
+pre,
+/* api docs code blocks */
+pre[class*="language-"]:has(code)[class*="language-"],
```
- moved plain-text related items here - uz static-files is example
- below was before the fix
![3_uz_pre_color_dark](https://github.com/user-attachments/assets/ff65bcb3-1589-4007-bb05-65c9662e74f8)
![3_uz_pre_color_light](https://github.com/user-attachments/assets/bb63ad0b-9e39-49c0-8682-34d87e920bce)

bjohansebas replied:
Probably need to finish #1746 first, right?

chrisdel101 replied:
Yeah, unless we want to update this after. Essentially all this would be removed then?

bjohansebas replied:
Part of that will be removed

bjohansebas replied:
I've finished the [PR](https://github.com/expressjs/expressjs.com/pull/1746), maybe you want to take a look so we can move forward with it.

chrisdel101 replied:
Yes will review it. Will require some significant changes over here after that, which is fine.

bjohansebas replied:
I think this is already resolved since it was completely fixed in another PR

bjohansebas commented:
Wow, it looks great, I'll review it very soon

Commit:
Clean up vars and fix small mistakes

bjohansebas committed:
Merge branch 'gh-pages' of github.com:expressjs/expressjs.com into design-system

bjohansebas commented:
@chrisdel101 i have fixed the conflicts

chrisdel101 commented:
@bjohansebas It's still a draft as I'm working fixing a few of the last details, but you can start a review as the remaining things are minor. Overall it's complete, but left as draft due to that. 

- I tried not to make any actual design changes, other than colors, and move things to variables - but there were a few changes that had to happen.
  - redid the overlay as shown above.
  - redid the `#top` arrow icon (in bottom left) that scrolls to the top of the page - its more pronounce on scroll and on hover.
  - added dark mode 'announcements' icon on index.
  - remove rules that did not exist - used Beautiful soup to search code base for things, and if not found I removed them. I tried to note this where possible with comments.  
  - Changed the way the desktop dropdown menu works, and the coloring. Now the top level of the drawer is off when drawer is open and hovering over a child item, and a bottom border was added for clarity on the top level e.g. Guide. I also plan to submit a PR for #1726 as I think in this img Guide should be bolded, not Getting Stared (which is bold since it's the page were currently on)
 
![image](https://github.com/user-attachments/assets/b5889e94-aa85-4bec-bda4-822616c57ff4)

- I'll delete all these "help" type comments that exist, which I left in to help with reviewing i.e. explaining why a change was made. 
- I tried to keep the number of colors to a minimum and so got rid of most of the colors that only occurred once or twice, and replaced with one of the system colors. It should be simple to make any changes to unliked colors, or add in new ones, using this same system. So, things do look different in places.
- In the values.css I noted which colors got replaced with which - this was mostly just helpful in dev for me, and I will delete these comments before merge.
I'll add anything else to this block as I complete my PR.

chrisdel101 commented on css/style.css:
```diff
@@ -755,22 +753,25 @@ footer {
-#menu em,
-#tags-side-menu em,
-#blog-side-menu em{
+#menu ul a.active {
```
Can we remove `#tags-side-menu` and `tags-menu.md`? I'd love that!

bjohansebas replied:
Yep, do it for now

chrisdel101 replied:
Done

bjohansebas commented on css/themes/dark-theme.css:
```diff
@@ -0,0 +1,80 @@
+@import "../values.css";
```
Why would we include the import? Just adding it like other files should be enough, right?

chrisdel101 replied:
REMOVED: Thought it was required to import to access vars, but seems it is not.

bjohansebas committed:
add more css to folders

bjohansebas committed:
remove announcemnet-icon variable

bjohansebas commented on css/values.css:
```diff
@@ -0,0 +1,70 @@
+.light-mode:root {
+  /* [#7e7e7e, dadada] */
+  --grey-fg: #aeaeae; /* %68 */
```
I see that we only use it for the description, we probably want to remove it and make all the text the same color, right?

chrisdel101 replied:
removed - changed to menu-grey.

bjohansebas committed:
remove announcement icon

bjohansebas committed:
Merge branch 'gh-pages' into design-system

bjohansebas commented:
We should still show when they are an inline code
![image](https://github.com/user-attachments/assets/1712ad9a-452d-4f31-83e2-ec5688997a78)
![image](https://github.com/user-attachments/assets/af4ec213-08b3-470c-aca7-342af5d20dac)

chrisdel101 committed:
Update merged changes to use vars - fix outliers
- move css to langs folder
- notate .no-release class

chrisdel101 committed:
Merge branch 'design-system' of github.com:chrisdel101/expressjs.com into design-system

chrisdel101 committed:
remove imports

Commit:
Merge remote-tracking branch 'upstream/gh-pages' into design-system

Commit:
Merge remote-tracking branch 'origin/design-system' into design-system

Commit:
Remove comments, tag menu

chrisdel101 commented on css/style.css:
```diff
@@ -200,35 +193,35 @@ li code {
   border: 4px solid transparent;
   background-clip: content-box;
-  background-color: hsl(0, 0%, 67%);
+  background-color: var(--grey-fg);
```
Just make sure to check scroll bar and hover.

chrisdel101 replied:
It's broken or something atm. Will need to check this now. Not sure what happened.

chrisdel101 replied:
There was some sort of overlay on scroll. Must have been caused by some other PR as it appears to be fixed.

chrisdel101 committed:
Remove grey-fg, fat-arrow style, comments

chrisdel101 commented:
> We should still show when they are an inline code ![image](https://private-user-images.githubusercontent.com/103585995/409657142-1712ad9a-452d-4f31-83e2-ec5688997a78.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg4NTMyNDgsIm5iZiI6MTczODg1Mjk0OCwicGF0aCI6Ii8xMDM1ODU5OTUvNDA5NjU3MTQyLTE3MTJhZDlhLTQ1MmQtNGYzMS04M2UyLWVjNTY4ODk5N2E3OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwNlQxNDQyMjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YTNlMjdkODc1NTdjZjhkMmY4OGZkMzhjZWJlODM0YWZkNGY5ZjJjYjQwMDYzODhjYWNhMWNhOWRiYTA3OTE3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.DCfPJjDorcvi-4jutiwfvAhBr39wK_Y52v1y1gRml3s) ![image](https://private-user-images.githubusercontent.com/103585995/409657212-af4ec213-08b3-470c-aca7-342af5d20dac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg4NTMyNDgsIm5iZiI6MTczODg1Mjk0OCwicGF0aCI6Ii8xMDM1ODU5OTUvNDA5NjU3MjEyLWFmNGVjMjEzLTA4YjMtNDcwYy1hY2E3LTM0MmFmNWQyMGRhYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwNlQxNDQyMjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yYWRiMmU2OGIxODYwN2VlNzk0ZGRlMzhjZmM0YTZiNmJkNGM5ODU3YjE4NTc0MzMxMmFlYWRlMzg1MzVlYjA0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.mlPrxZS1mWNuUHcoa0JlbjB1o3ScR1YVbWPpEwOgGdM)

I looked compared the blog page to prod and I don't see any visible difference besides some small line height spacing. Is this still and issue?
If yes, can you be more specific?

chrisdel101 commented:
~~I noticed a problem with the yaml rendering here. Wasn't sure the best way to fix.~~ I changed it to `markdown` and it looks good now.
![image](https://github.com/user-attachments/assets/55034d1c-ee56-4639-a992-1abeb0f4d415)
And there spaces at the start of lines. Not sure why. It's like this in prod too. Was it like this with Prism? 
![space](https://github.com/user-attachments/assets/ff06f149-7d33-4c86-99cd-eb8078735f34)

chrisdel101 committed:
Change yaml tag to markdown

carlosstenzel commented:
It is in every block of code, including the home page.
![Captura de Tela 2025-02-06 às 13 09 56](https://github.com/user-attachments/assets/e6b5ec01-874e-4bfc-9339-acb9509ecd05)

we can see this padding, it seems to resolve

![Captura de Tela 2025-02-06 às 13 10 52](https://github.com/user-attachments/assets/6c09e8d3-b6af-4d1b-a94f-5f4a95aed4d7)

chrisdel101 committed:
Merge branch 'gh-pages' into design-system

carlosstenzel commented:
@chrisdel101 
 there is space in the code tag

chrisdel101 commented:
@carlosstenzel Where is the space? I can't see it.

carlosstenzel commented:
![image](https://github.com/user-attachments/assets/caeb7173-feab-43d5-9ef5-3dc8058a3280)

bjohansebas commented:
It wasn't intentional, although I already know how to fix it because I had already solved it before due to the same error. I recommend creating an issue instead of discussing it here, since this is not related to this PR.

chrisdel101 commented:
Okay, isn't this the same thing I noted above and you told me to fix using padding?
I thought U meant something else.

carlosstenzel commented:
> It wasn't intentional, although I already know how to fix it because I had already solved it before due to the same error. I recommend creating an issue instead of discussing it here, since this is not related to this PR.

Sorry guys, we left this adjustment for another PR

bjohansebas commented on css/style.css:
```diff
@@ -1,27 +1,21 @@
-  --warn-accent: rgb(138, 15, 58);
-  --code-bg: #f0f1f3;
-}
+@import "./values.css";
```
Let's do this, another link tag in the head. I'm sure we shouldn't use import in CSS, it's considered bad practice if I remember correctly, unless we use a preprocessor like Sass.

bjohansebas commented on css/style.css:
```diff
@@ -1,27 +1,21 @@
-/* general */
+/*
+ Copyright (c) 2016 Express Contributors
+ License: MIT
+*/
```
please remove this

bjohansebas commented on css/style.css:
```diff
@@ -1321,6 +1289,9 @@ blockquote {
   #navbar .submenu.open {
     flex-direction: column;
     align-items: initial;
+    > a {
```
```suggestion
    & a {

```

chrisdel101 replied:
But `>` is direct descendant and `&` is just descendant, no?
Seems to still work b/c there is only one descendant but > is what we really want I think.

bjohansebas replied:
Ah, okay, I thought it was a typo.

bjohansebas commented on css/themes/light-theme.css:
Is there any reason why we can't delete this file and put those styles directly in `styles.css`, applying them directly to their respective tags?

```css
#top {
opacity: .2;
  line-height: 0;
  border-radius: 2px;
  position: fixed;
  bottom: 15px;
  // .....
  background: var(--box-fg);
  
}
```

chrisdel101 replied:
Because they are specific to light mode. Not every single change could be binary, but i tried my best. There are a still a few that are totally seperate, like these. And they cannot be put in style since the entire style.css file is scoped to light or dark mode.

chrisdel101 replied:
Sorry, I think the answer is actually yes we can do it for this one actually.

EDIT: Wait, no, only light mode has a dark background where at the dark-mode does not (it looks bad just flipping)  - so after all it cannot be moved. 
![Screenshot 2025-02-06 at 8 17 00 PM](https://github.com/user-attachments/assets/84e6180b-5f3c-4328-8124-798535f23c67)
![Screenshot 2025-02-06 at 8 16 44 PM](https://github.com/user-attachments/assets/2e23aa34-b8f4-4fcb-b5dc-7f4a52d3cf27)

bjohansebas commented on css/values.css:
```diff
@@ -0,0 +1,51 @@
+.light-mode:root {
```
I meant setting those variables directly in `:root` to make them even more global.
```suggestion
:root {
```

chrisdel101 replied:
Those are scoped. Removing those classes means the styles are not applied to the right themes.
Unless you are just saying?

chrisdel101 replied:
Just to reiterate - root sets variables globally, and this just scopes it to have a class. I rearranged the order for clarity.

bjohansebas replied:
So we would have to add the `light-mode` class to the `<html>` tag, as you initially suggested, since there may be users who disable JavaScript, and the page would look like this:
![image](https://github.com/user-attachments/assets/24b146aa-0744-4762-8979-e4fac0ae2493)

chrisdel101 replied:
Put the light-mode tag back in.

Commit:
Merge remote-tracking branch 'upstream/gh-pages' into design-system

Commit:
Make suggesed changes

Commit:
Move drop down styles to style.css

Commit:
Finish drop down fix

chrisdel101 commented:
Hey @bjohansebas, I made the changes and left comments re: what I did not change. I don't think anything else can be moved rules-wise between files now.

I think I've moved over everything I could now into style.css

bjohansebas committed:
Unify the overlay background

bjohansebas commented on css/style.css:
```diff
@@ -360,21 +334,26 @@ pre code {
+}
+
+#top:hover {
+  border: 2px solid var(--border); 
```
```suggestion
  border: 1px solid var(--border); 
```

I would suggest not increasing the border size. The effect it creates when hovering and expanding doesn't look natural.

chrisdel101 replied:
Undid the changes back to 1px

bjohansebas commented on css/style.css:
```diff
@@ -360,21 +334,26 @@ pre code {
 }
 
 #top {
+  opacity: .2;
```
If it's at the top of the page, we shouldn't show that element, which is why its opacity is set to 0.

chrisdel101 replied:
Removed it.
I can't remember why I would have added this. Maybe it was something I forgot to remove?

bjohansebas committed:
Update code background color in variables.css

bjohansebas commented:
Great job @chrisdel101! I've left the last two comments, and I can give my approval. btw, I updated the color of the code examples.

Commit:
Make requested changes - comments in JS

Commit:
Merge remote-tracking branch 'origin/design-system' into design-system

chrisdel101 commented:
@bjohansebas Hey, I made the newest set of requested changes. I added a few comments to the theme JS file here for clarity, but did not change any JS but I changed the names of a few functions (only the names).
Let me know if you find anything else. Thanks for all the help with this review!

bjohansebas committed:
readd opacity to #top class

bjohansebas reviewed (approved):
LGTM, I'm going to wait a day so more people can review it

chrisdel101 commented:
@bjohansebas Just a heads up, I'm going to be away for awhile for medical reasons - maybe like a month. If this branch gets out of date you might need to take over the updates :)

bjohansebas committed:
Refactor #top button styles and remove light theme CSS

bjohansebas commented:
@chrisdel101 Great job and take care!

chrisdel101 commented:
@bjohansebas Once again thanks for all your help on this, and for the thorough review.
